### PR TITLE
Add convert to bool if string or number receive for boolean value

### PR DESCRIPTION
**Pull request by danielailie**
Add convert to bool before creating the boolean value to cover the scenarios where boolean value is passed as string or number.
This is a follow up at this issue https://github.com/multiversx/mx-sdk-js-core/issues/258

### CODE REVIEW

File: src/smartcontracts/nativeSerializer.spec.ts [hunk 1]
```diff
@@ -610,4 +610,59 @@ describe("test native serializer", () => {
...
+                            type: "bool",
+                        },
+                    ],
+                    outputs: [],
+                },
+            ],
+        }).getEndpoint("foo");
+
+        let typedValues = NativeSerializer.nativeToTypedValues(
+            [true, "true", "TRUE", 1, false, "false", "falseString", 0, 5],
```
Comment (andreibancioiu) [inline src/smartcontracts/nativeSerializer.spec.ts:654]: :rocket:

Comment (andreibancioiu) [inline src/smartcontracts/nativeSerializer.spec.ts:654]: `"falseString"` can be `"foobar"` - test is thus more explicit about arbitrary strings.

File: src/smartcontracts/nativeSerializer.spec.ts [hunk 2]
```diff
@@ -610,4 +610,59 @@ describe("test native serializer", () => {
             variadic: false,
         });
     });
+
+    it.only("should accept a mixed of values for boolen type", async () => {
```
Comment (andreibancioiu) [inline src/smartcontracts/nativeSerializer.spec.ts:614]: Slipped "only".

File: src/smartcontracts/nativeSerializer.ts [hunk 3]
```diff
@@ -296,7 +296,8 @@ export namespace NativeSerializer {
             return new AddressValue(convertNativeToAddress(native, errorContext));
         }
         if (type instanceof BooleanType) {
-            return new BooleanValue(native);
+            const boolValue = native.toString().toLowerCase() === "true" || native.toString() === "1";
```
Comment (andreibancioiu) [inline src/smartcontracts/nativeSerializer.ts:299]: :rocket:

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@multiversx/sdk-core",
-  "version": "13.2.2",
+  "version": "13.2.3",
```
Comment (andreibancioiu) [inline package.json:3]: Let's bump the minor version, so that developers take this update intentionally (not as a hotfix) - `v13.3.0`.

**Review by andreibancioiu — APPROVED** [review]

File: src/smartcontracts/nativeSerializer.ts (hunk 3 above)
Comment (ccorcoveanu) [reply src/smartcontracts/nativeSerializer.ts:299]: Wouldn't this be more appropriate in the "isTrue" evaluation?

**Review by ccorcoveanu — APPROVED** [review]

**Pull request merged**